### DISCUSSION TITLE
📣 Announcement: GraphQL Java release policy

### DISCUSSION THREAD

**Discussion by dondonz** [graphql-java/graphql-java#3138] [General]
Hello everyone,

We’re formalising our release schedule to give the community a better idea of when to expect releases, what will be contained within them, and when important fixes will be backported.

See the full details on our blog: https://www.graphql-java.com/blog/release-policy

**Comment by anbusampath** [comment]
GH milestones names as months like [2023 April](https://github.com/graphql-java/graphql-java/milestone/48) as well as number [19.4](https://github.com/graphql-java/graphql-java/milestone/52). what is the difference?

**Reply by dondonz** [reply]
Well spotted, I had forgotten to change "2023 April" to 20.1 with today's release. I've made that change now.

We've moved to quarterly releases, but it is still possible that an urgent security patch or important bugfix means we have an extra release, before the next quarter's date.

To avoid constantly renaming the milestone numbers when we have special security or bugfix releases, we decided to have the future milestones based on time. When milestones are released, we'll change the milestone's name to be the released version name.

**Reply by anbusampath** [reply]
> To avoid constantly renaming the milestone numbers when we have special security or bugfix releases, we decided to have the future milestones based on time. When milestones are released, we'll change the milestone's name to be the released version name.

that makes sense. Thanks.
